File: src/main/java/com/pr/gradle/task/ExecFork.java
package com.pr.gradle.task;

import java.util.ArrayList;
import java.util.List;

import org.gradle.api.GradleException;

public class ExecFork extends AbstractExecFork {
  public String commandLine;

  public void doValidation() {
    if (commandLine == null) {
      throw new GradleException(ExecFork.class.getSimpleName() + " task '" + getName() + "' must specify a commandLine");
    }
  }

  @Override
  public List<String> getProcessArgs() {
    List<String> processArgs = new ArrayList<>();
    processArgs.add(commandLine);
    processArgs.addAll(args);
    return processArgs;
  }
}

File: src/main/kotlin/com/github/psxpaul/task/ExecFork.kt
package com.github.psxpaul.task

import org.gradle.api.tasks.InputFile
import org.gradle.process.ProcessForkOptions
import org.gradle.process.internal.JavaForkOptionsFactory
import javax.inject.Inject

/**
 * A task that will run a command in a separate process, optionally
 * writing stdout and stderr to disk, and waiting for a specified
 * port to be open.
 *
 * @see AbstractExecFork
 * @see ProcessForkOptions for all available configuration options
 */
open class ExecFork @Inject constructor(forkOptionsFactory: JavaForkOptionsFactory) : AbstractExecFork(),
        ProcessForkOptions by forkOptionsFactory.newJavaForkOptions() {

    /**
     * The path to the executable to run
     * @deprecated Use #executable instead
     */
    @get:InputFile
    var commandLine: String?
        get() = executable
        set(value) {
            executable = value
        }

    override fun getProcessArgs(): List<String>? {
        val processArgs: MutableList<String> = mutableListOf()
        processArgs.add(executable!!)
        processArgs.addAll(args.map(CharSequence::toString))
        return processArgs
    }

}